backend/src/main/java/org/eclipse/tractusx/puris/backend/common/edc/logic/util/EdcRequestBodyBuilder.java

```diff
@@ -419,19 +420,39 @@ public JsonNode buildSubmodelRegistrationBody(String assetId, String endpoint, S
         propertiesObject.set("aas-semantics:semanticId", semanticIdObject);
         semanticIdObject.put("@id", semanticId);
         body.set("privateProperties", MAPPER.createObjectNode());
+        body.set("dataAddress", createDataAddressObject(endpoint, "false"));
+        return body;
+    }
 
+    public JsonNode buildNotificationRegistrationBody(String assetId, String endpoint, String semanticId) {
+        var body = getAssetRegistrationContext();
+        body.put("@id", assetId);
+        var propertiesObject = MAPPER.createObjectNode();
+        body.set("properties", propertiesObject);
+        var dctTypeObject = MAPPER.createObjectNode();
+        propertiesObject.set("dct:type", dctTypeObject);
+        dctTypeObject.put("@id", "cx-taxo:DemandAndCapacityNotificationApi");
+        propertiesObject.put("cx-common:version", "1.0");
+        var semanticIdObject = MAPPER.createObjectNode();
```

Comment:
please remove semanticId object as it's not needed per standard. Also `privateProperties` are not needed.